assets/js/state/sapSystems.test.js

```diff
@@ -113,4 +115,54 @@ describe('SAP Systems reducer', () => {
 
     expect(sapSystemsReducer(initialState, action)).toEqual(expectedState);
   });
+
+  it('should upsert database instances', () => {
+    const changedIndex = 0;
+
+    const initialState = {
+      databaseInstances: databaseInstanceFactory.buildList(2),
+    };
+
+    const updatedInstance = {
+      ...initialState.databaseInstances[changedIndex],
+      instance_hostname: 'my_name_has_changed',
+    };
+    const newInstance = databaseInstanceFactory.build();
+    const newInstances = [updatedInstance, newInstance];
+
+    const action = upsertDatabaseInstances(newInstances);
+
+    const expectedState = {
+      databaseInstances: initialState.databaseInstances
```

Comment:
Really complex...
Cannot we have a fixed expected thing so everyone can understand in a first glance what's going on?